bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetDeviceHandler.java

```diff
@@ -0,0 +1,295 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JuiceNetDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetDeviceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetDeviceHandler.class);
+
+    protected String name = "";
+    protected String id = "";
+    protected String token = "";
+
+    protected long targetTimeTou = 0;
+    protected long lastInfoTimestamp = 0;
+
+    JuiceNetApi.JuiceNetApiDeviceStatus deviceStatus = new JuiceNetApi.JuiceNetApiDeviceStatus();
+    JuiceNetApi.JuiceNetApiInfo deviceInfo = new JuiceNetApi.JuiceNetApiInfo();
+    JuiceNetApi.JuiceNetApiTouSchedule deviceTouSchedule = new JuiceNetApi.JuiceNetApiTouSchedule();
+    JuiceNetApi.JuiceNetApiCar deviceCar = new JuiceNetApi.JuiceNetApiCar();
+
+    // This holds the last time the info was updated and is compared everytime we get device status to determine
+    // if the deviceInfo should be updated. Set the time in the past to ensure info is updated on initial call.
+    protected ZonedDateTime infoTimestamp = ZonedDateTime.of(2000, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+
+    public JuiceNetDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("JuiceNetDeviceHandler:initialize");
+        Map<String, String> properties = editProperties();
+
+        name = Objects.requireNonNull(properties.get(DEVICE_PROP_NAME));
+        id = Objects.requireNonNull(properties.get(DEVICE_PROP_UNIT_ID));
+        token = Objects.requireNonNull(properties.get(DEVICE_PROP_TOKEN));
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            try {
+                queryDeviceStatusAndInfo();
+            } catch (JuiceNetApiException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+                return;
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+                return;
+            } catch (InterruptedException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+                return;
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+        });
+    }
+
+    protected JuiceNetApi getApi() {
+        Bridge bridge = Objects.requireNonNull(getBridge());
+        BridgeHandler handler = Objects.requireNonNull(bridge.getHandler());
+
+        return ((JuiceNetBridgeHandler) handler).getApi();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand");
```

Comment:
If this is really needed, I'd propose TRACE log level and to include channelUID and command.